Add docs for Alluxio file system

jja725 opened the pull request:
## Description

Add docs for Alluxio file system.

## Additional context and related issues

Docs for https://github.com/trinodb/trino/pull/21603

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

mosabua commented on docs/src/main/sphinx/object-storage.md:
```diff
@@ -60,6 +60,7 @@ compatible replacements:
 * [](/object-storage/file-system-azure)
 * [](/object-storage/file-system-gcs)
 * [](/object-storage/file-system-s3)
+* [](/object-storage/file-system-alluxio)
```
Not sure if we want it here .. wdyt @electrum ?

mosabua commented on docs/src/main/sphinx/object-storage/file-system-alluxio.md:
```diff
@@ -0,0 +1,39 @@
+
+The Alluxio client in `alluxio-site.properties` configures the connection to the
+Alluxio cluster. The default location of the `alluxio-site.properties` is
+`/opt/alluxio/conf`. Follow the [Alluxio client configuration
```
Is this location configurable .. if yes .. we should add to the table. Also .. I assume this file is needed on the Trino coordinator and all workers .. right?

jja725 replied:
The location is only configurable through Trino side at the moment. We can make it configurable in a later PR. Yes it's need to be on every process.

jja725 replied:
sorry for the typo, the location is not configurable at the moment. We can make it configurable at Trino side in a later PR

mosabua replied:
Sounds good

mosabua commented:
I did a bunch of updates for grammar and setup so the docs actually build and work. Left a few questions but in general this looks good. I am wondering if we should have a minimal example of the properties file though .. and then link to the Alluxio docs for me. 

Also please review all content I updated and update as needed and push further changes @jja725

jja725 commented:
> I did a bunch of updates for grammar and setup so the docs actually build and work. Left a few questions but in general this looks good. I am wondering if we should have a minimal example of the properties file though .. and then link to the Alluxio docs for me.
> 
> Also please review all content I updated and update as needed and push further changes @jja725

Added example property file. The updates LGTM, thanks for collaborating on this!

jja725 commented on docs/src/main/sphinx/object-storage/file-system-alluxio.md:
```diff
@@ -0,0 +1,44 @@
+## Alluxio client configuration
+
+The Alluxio client in `alluxio-site.properties` configures the connection to the
+Alluxio cluster. The default location of the `alluxio-site.properties` is
```
resolved

mosabua reviewed (approved):
Looks good now. Thank you @jja725
